### PR TITLE
XL release OpsGenie plugin

### PR DESCRIPTION
**Pull request by mehrat**
Plugin has following tasks :-
* Create alert in OpsGenie
* Execute custom action in OpsGenie
* Close alert in OpsGenie
* Delete alert in OpsGenie
* Snooze alert in OpsGenie
* Enable integration in OpsGenie
* Disable integration in OpsGenie

### CODE REVIEW

File: src/main/resources/synthetic.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    <type type="opsgenie.config" extends="configuration.HttpConnection">
+        <property name="scriptLocation" default="opsgenie_xlr/test_connection.py" hidden="true" />
+        <property name="apiKey" required="true" password="true" description="OpsGenie API key."/>
+        <property name="url" hidden="true" required="false" default="https://api.opsgenie.com/v1/json"/>
+        <property name="username" hidden="true" required="false" />
+        <property name="password" hidden="true" required="false" />
+    </type>
+
+    <type type="opsgenie.task" extends="xlrelease.PythonScript" virtual="true">
+        <property name="OpsGenieConfig" label="OpsGenie Config" referenced-type="opsgenie.config" kind="ci" category="input" description="OpsGenie API configuration."/>
```
Comment (saraswat2385) [inline src/main/resources/synthetic.xml]: name="OpsGenieConfig" dosen't follow standard naming convention

File: src/main/resources/opsgenie_xlr/test_connection.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    response = AlertApi().list_alerts(
+            limit=5,
+            query='status: open',
+            order='desc',
+            sort='createdAt')
+    if not response.data:
+        sys.exit(1)
+
+except ApiException as err:
+    print("Exception when calling AlertApi->list_alerts: %s\n" % err)
```
Comment (saraswat2385) [inline src/main/resources/opsgenie_xlr/test_connection.py]: error msg is not appropriate

File: src/main/resources/opsgenie_xlr/api_client.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        SwaggerConf.ssl_ca_cert = os.environ.get('REQUESTS_CA_BUNDLE')
+        configuration.swag_conf = SwaggerConf
+
+        if self.proxy_host:
+            proxy_uri = urlparse(self.proxy_host)
+            proxy_config = ProxyConfiguration(self.proxy_host, self.proxy_port, self.proxy_username, self.proxy_password, proxy_uri.scheme)
+            configuration.proxy_config = proxy_config
+
+
+    def _disable_https_warnings(self):
```
[the remark this answers is not in the document]
Comment (mehrat) [reply src/main/resources/opsgenie_xlr/api_client.py]: corrected

File: src/main/resources/opsgenie_xlr/close_alert.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+client.setup_opsgenie_client()
+
+body = CloseAlertRequest(
+        source=source,
+        user=user,
+        note=note)
+try:
+    response = AlertApi().close_alert(identifier=alertIdentifier, identifier_type=alertIdentifierType, body=body)
+    if response.request_id:
+        print "Alert %s closed." % (alertIdentifier)
```
[the remark this answers is not in the document]
Comment (mehrat) [reply src/main/resources/opsgenie_xlr/close_alert.py]: done

**Comment by mehrat** [comment]
@saraswat2385 Review comments incorporated, please review and merge

File: src/main/resources/synthetic.xml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+           xsi:schemaLocation="http://www.xebialabs.com/deployit/synthetic synthetic.xsd">
+
+    <type type="opsgenie.config" extends="configuration.HttpConnection">
+        <property name="scriptLocation" default="opsgenie_xlr/test_connection.py" hidden="true" />
+        <property name="apiKey" required="true" password="true" description="OpsGenie API key."/>
+        <property name="url" hidden="true" required="false" default="https://api.opsgenie.com/v1/json"/>
+        <property name="username" hidden="true" required="false" />
+        <property name="password" hidden="true" required="false" />
+        <property name="authenticationMethod" kind="enum" enum-class="com.xebialabs.xlrelease.domain.configuration.HttpConnection$AuthenticationMethod" hidden="true" required="false" />
+        <!--<enum-values>-->
```
Comment (saraswat2385) [inline src/main/resources/synthetic.xml]: Remove commented code

Comment (saraswat2385) [inline src/main/resources/synthetic.xml]: remove commeted code

**Pull request merged**